IDinsight/aaq-core
# Display API key info in front-end

Reviewer: @sidravi1 
Estimate: 30mins

---

## Ticket

Fixes: [534](https://idinsight.atlassian.net/browse/AAQ-534)

## Description

### Goal

### Changes

- Added first 5 chars and changed date to DB
- Added user GET endpoint
- Add to front-end
- Handle date correctly
- Use existence of current key as a condition show the right message/flow when creating a new key.
- Add pytest for user GET endpoint

## How has this been tested?

- run pytests
- run dev setup. to see loading animation/behaviour, you might have to add

```
      // 1s wait
      await new Promise((resolve) => setTimeout(resolve, 1000));
```
before
```
      const data = await apiCalls.createNewApiKey(token!);
```
so you can see the loading.
- run docker compose

## To-do before merge (optional)

## Checklist

Fill with `x` for completed.
